### ISSUE TITLE
Claim with `ensures` clause causes normalization error

### ISSUE THREAD

**Issue by sskeirik** [kframework/kore#2221]
## Repro information

Here is the command run:

```
kompile --backend haskell --gen-bison-parser list-sizer.k && kprove -m LIST-SIZER spec.k
```

CLI Error:

```
kore-exec: [151389] Error (ErrorException):
    Assertion failed
    CallStack (from HasCallStack):
      assert, called at src/Kore/Internal/Conditional.hs:459:7 in kore-0.32.0.0-FCSM5feNLSFEuq2Thk9UtU:Kore.Internal.Conditional
      assertNormalized, called at src/Kore/Internal/SideCondition.hs:219:20 in kore-0.32.0.0-FCSM5feNLSFEuq2Thk9UtU:Kore.Internal.SideCondition
      assertNormalized, called at src/Kore/Internal/SideCondition.hs:155:5 in kore-0.32.0.0-FCSM5feNLSFEuq2Thk9UtU:Kore.Internal.SideCondition
    Please file a bug report: https://github.com/kframework/kore/issues
```

Version Info:

```
kompile --version
RV-K version 1.0-SNAPSHOT
Git revision: 527733d
Git branch: master
Build date: Thu Oct 22 11:49:08 CDT 2020
```

## Kore Dump

Available here: [kore-exec.tar.gz](https://github.com/kframework/kore/files/5424378/kore-exec.tar.gz)

## Spec Overview

Here is the semantics:

```
module LIST-SIZER-SYNTAX                               
  imports INT-SYNTAX                                   
  syntax MyList ::= List{Int, ""}                      
endmodule                                              
                                                       
module LIST-SIZER                                      
  imports LIST-SIZER-SYNTAX                            
  imports INT                                          
  imports LIST                                         
                                                       
  syntax List ::= toList(MyList) [function, functional]
  // --------------------------------------------------
  rule toList(I L:MyList) => ListItem(I) toList(L)     
  rule toList(   .MyList) => .List                     
                                                       
  configuration <k> toList($PGM:MyList) </k>           
                <size> 0 </size>                       
                                                       
  rule <k> .List => .K </k>                            
                                                       
  rule <k> ListItem(_E) L:List => L:List </k>          
       <size> N => N +Int 1 </size>                    
endmodule                                              
```

Here is the spec file:

```
module SPEC                              
  imports LIST-SIZER                     
                                         
  claim <k> ListItem(_) _:List => .K </k>
        <size> 0 => ?N </size>           
     ensures ?N >=Int 0                 
endmodule
```

**Comment by sskeirik** [comment]
Whoops. I made a very dumb mistake in formatting my spec. Let me check that and reopen it if that doesn't fix the problem.

**Comment by sskeirik** [comment]
OK, I fixed the silly issue and was still able to generate the same error. Sorry for the bad initial filing.

**Comment by ttuegel** [comment]
Per our conversation, this seems to be caused by the `ensures` clause in the claim. This is surprising, because the `ensures` clause is not used (as far as I know) in a `SideCondition`.

**Comment by sskeirik** [comment]
I have a simplified version of the example which causes the same issue. This example has no special constructors except for `Int` and no `ensures` clause.

Command run:

```
kompile --backend haskell  list-sizer.k
kprove -m LIST-SIZER invalid-size.spec
```

CLI Output:

```
kore-exec: [142675] Error (ErrorException):
    Assertion failed
    CallStack (from HasCallStack):
      assert, called at src/Kore/Internal/Conditional.hs:459:7 in kore-0.32.0.0-FCSM5feNLSFEuq2Thk9UtU:Kore.Internal.Conditional
      assertNormalized, called at src/Kore/Internal/SideCondition.hs:219:20 in kore-0.32.0.0-FCSM5feNLSFEuq2Thk9UtU:Kore.Internal.SideCondition
      assertNormalized, called at src/Kore/Internal/SideCondition.hs:155:5 in kore-0.32.0.0-FCSM5feNLSFEuq2Thk9UtU:Kore.Internal.SideCondition
    Please file a bug report: https://github.com/kframework/kore/issues
```

Kore Dump:  [kore-exec.tar.gz](https://github.com/kframework/kore/files/5431595/kore-exec.tar.gz)

Semantics:

```
module LIST-SIZER-SYNTAX
  imports INT-SYNTAX
  syntax MyList ::= ".MyList"
                  | Int MyList
endmodule

module LIST-SIZER
  imports LIST-SIZER-SYNTAX
  imports INT
  imports LIST

  configuration <k> $PGM:MyList </k>
                <size> 0 </size>

  rule <k> .MyList => .K </k>

  rule <k> _E L:MyList => L </k>
       <size> N => N +Int 1 </size>
endmodule
```

Spec:

```
module INVALID-SIZE
  imports LIST-SIZER

  claim <k> _:MyList => .K </k>
        <size> 0 </size>
endmodule
```

**Comment by ttuegel** [comment]
Thanks for minimizing this! Is it blocking?

**Comment by sskeirik** [comment]
I was able to reproduce this bug in the Michelson semantics on our important proof. Here is the Kore dump:
[kore-exec.tar.gz](https://github.com/kframework/kore/files/5459997/kore-exec.tar.gz).

**Comment by ttuegel** [comment]
I believe this would be fixed by #2174.
